templates/default/etc/systemd/system/knotx.service.erb

```diff
@@ -1,30 +1,30 @@
 [Unit]
```

Comment:
btw, I recently found interesting stuff about how to do a gracefull shutdown of java process in system.d, see: https://stegard.net/2016/08/gracefully-killing-a-java-process-managed-by-systemd/ - maybe it's worth to look at

Reply:
Good point! Indeed `SIGTERM` sounds to be definitely a better way, as knot.x stops with the following message in this case:

```
2018-04-25 13:34:14.274 [vert.x-eventloop-thread-5] INFO  i.k.k.t.HandlebarsKnotVerticle - Stopping <HandlebarsKnotVerticle>
```

Reply:
Currently the service returns the following when stopped:

```
$ systemctl status knotx-primary
● knotx-primary.service - knotx-primary
   Loaded: loaded (/etc/systemd/system/knotx-primary.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Wed 2018-04-25 11:39:57 UTC; 3s ago
 Main PID: 5149 (code=exited, status=143)
```

Reply:
According to systemd [docs](https://www.freedesktop.org/software/systemd/man/systemd.kill.html) default `KillSignal` is set to `SIGTERM`. That confirms what I observed above (exit code `143`).

When I reduced `logback.xml` to

```
<configuration>
    <property name="LOG_PATH" value="/var/log/knotx/primary"/>

    <include resource="io/knotx/logging/logback/defaults.xml" />
    <include resource="io/knotx/logging/logback/file-appender.xml" />
    <include resource="io/knotx/logging/logback/access.xml" />

    <root level="INFO">
        <appender-ref ref="FILE" />
    </root>
</configuration>
```

I can see 

```
2018-04-25 12:51:26.374 [vert.x-eventloop-thread-1] INFO  i.k.k.t.HandlebarsKnotVerticle - Stopping <HandlebarsKnotVerticle>
```

in the kont.x log file when service gets stopped.

The only thing we should add is `SuccessExitStatus=143` to let systemd know that such exit code means successful shutdown. Just pushed that and indeed it looks better

```
$ systemctl status knotx-primary
● knotx-primary.service - knotx-primary
   Loaded: loaded (/etc/systemd/system/knotx-primary.service; enabled; vendor preset: disabled)
   Active: inactive (dead) since Wed 2018-04-25 12:55:33 UTC; 1s ago
 Main PID: 12992 (code=exited, status=143)
```